mlflow/pyfunc/model.py

```diff
@@ -228,17 +230,39 @@ def _save_model_with_class_artifacts_params(
     if artifacts:
         saved_artifacts_config = {}
         with TempDir() as tmp_artifacts_dir:
-            tmp_artifacts_config = {}
             saved_artifacts_dir_subpath = "artifacts"
+            hf_prefix = "hf:/"
             for artifact_name, artifact_uri in artifacts.items():
-                tmp_artifact_path = _download_artifact_from_uri(
-                    artifact_uri=artifact_uri, output_path=tmp_artifacts_dir.path()
-                )
-                tmp_artifacts_config[artifact_name] = tmp_artifact_path
-                saved_artifact_subpath = posixpath.join(
-                    saved_artifacts_dir_subpath,
-                    os.path.relpath(path=tmp_artifact_path, start=tmp_artifacts_dir.path()),
-                )
+                if artifact_uri.startswith(hf_prefix):
+                    try:
+                        from huggingface_hub import snapshot_download
+
+                        repo_id = artifact_uri[len(hf_prefix) :]
+                        snapshot_location = snapshot_download(
+                            repo_id=repo_id,
+                            cache_dir=os.path.join(
+                                path, saved_artifacts_dir_subpath, artifact_name
+                            ),
+                        )
+                        saved_artifact_subpath = os.path.relpath(path=snapshot_location, start=path)
+                    except ImportError as e:
+                        raise MlflowException(
+                            "Failed to import huggingface_hub. Please install huggingface_hub "
+                            f"to log the model with artifact_uri {artifact_uri}. Error: {e}"
+                        )
```

Comment:
The current code makes me wonder what throws `ImportError`. Instead, we can do this:

```python
                    try:
                        from huggingface_hub import snapshot_download
                    except ImportError as e:
                        raise MlflowException(
                            "Failed to import huggingface_hub. Please install huggingface_hub "
                            f"to log the model with artifact_uri {artifact_uri}. Error: {e}"
                        )
```